kbs/src/plugins/implementations/resource/backend.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use anyhow::{bail, Context, Error, Result};
+use serde::Deserialize;
+
+use super::local_fs;
+
+type RepositoryInstance = Arc<dyn StorageBackend>;
+
+/// Interface of a `Repository`.
+#[async_trait::async_trait]
+pub trait StorageBackend: Send + Sync {
+    /// Read secret resource from repository.
+    async fn read_secret_resource(&self, resource_desc: ResourceDesc) -> Result<Vec<u8>>;
+
+    /// Write secret resource into repository
+    async fn write_secret_resource(&self, resource_desc: ResourceDesc, data: &[u8]) -> Result<()>;
+}
+
+#[derive(Debug, Clone)]
+pub struct ResourceDesc {
+    pub repository_name: String,
+    pub resource_type: String,
+    pub resource_tag: String,
+}
+
+impl TryFrom<&str> for ResourceDesc {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self> {
```

Comment:
this is a bit confusing. maybe use a regex to specify a legal path instead?

e.g.certain path_parts can have `.` and `..` in their name (the tag), but if we have 2 slashes and a "default" repository then resource tag isn't allowed to have those characters, I think

I would add a couple of unit test cases, so we have specified the desired behavior

Reply:
Cool. Regex is a nice idea. 

> e.g.certain path_parts can have . and .. in their name (the tag), but if we have 2 slashes and a "default" repository then resource tag isn't allowed to have those characters, I think

I use a regex to force the `repo`s, `type`s and `tag`s can only be `[A-Za-z0-9_-]+`. I think the restrict would be fine to avoid unknown potential risks?